nodejs/week3/homework/warmup/index.js

```diff
@@ -0,0 +1,59 @@
+const express = require("express");
+const app = express();
+app.get("/", (req, res) => res.send("nodejs week3 homework"));
+
+//Let's again create a calculator. This time the calculator will support 4 features(+ - / *)
+//All the numbers that should be inputted should be gotten from the query parameters: calculator?firstParam=1&secondParam=2...
+
+app.get("/add", async (req, res) => {
+  const params = req.query;
+  const values = Object.values(params).flat();
+  const add = values.reduce(
+    (firstParam, secondParam) => firstParam + secondParam
+  );
+  res.send(`${add}`);
+  if (!add) {
+    res.status(400).send("Params must be numbers");
+  } else console.log(Subtraction);
+  res.json(Subtraction);
+});
+
+app.get("/Subtraction", (req, res) => {
+  const params = req.query;
+  const values = Object.values(params).flat();
+  const Subtraction = values.reduce(
+    (firstParam, secondParam) => firstParam / secondParam
+  );
+  res.send(`${Subtraction}`);
+  if (!Subtraction) {
+    res.status(400).send("Params must be numbers");
+  } else console.log(Subtraction);
+  res.json(Subtraction);
+});
+app.post("/Division", async (req, res) => {
+  const params = req.query;
+  const values = Object.values(params).flat();
+  const Division = values.reduce(
+    (firstParam, secondParam) => firstParam / secondParam
+  );
+  res.send(`${Division}`);
+  if (!Division) {
+    res.status(400).send("Params must be numbers");
+  } else console.log(Division);
+  res.json(Division);
+});
+app.post("/multiply", async (req, res) => {
+  const params = req.query;
+  const values = Object.values(params).flat();
+  const multiply = values.reduce(
+    (firstParam, secondParam) => firstParam * secondParam
+  );
+  res.send(`${multiply}`);
+  if (!multiply) {
+    res.status(400).send("Params must be numbers");
+  } else console.log(multiply);
+  res.json(multiply);
+});
```

Comment:
This is a post request, but you're still accessing the query parameters, which are passed in as part of the url.
You need to read the parameters from the request body!
Here is some more information about the difference between GET and POST requests:
https://www.guru99.com/difference-get-post-http.html and https://www.postman.com/ is a useful tool to make POST requests and test your api.
Also, here is how you can access the body parameters in nodejs using express:
https://nodejs.dev/learn/get-http-request-body-data-using-nodejs

Reply:
thanks a lot I will read that and then Implement changes

Reply:
Unfortunately I couldn't fix it really can you help me with it little more?

Reply:
Absolutely! So, regarding GET and POST requests:

- An http request consists of a 'request line', a 'headers' section and a 'body' section.
- When you're using GET requests, you can only send in parameters as part of the url: eg http:localhost:3000/calculator/add?param=1&anotherparam=2. So, when you are entering a url to your browser address bar, you're basically sending in GET requests to a server! 
- When you're using POST requests you are sending parameters as part of the 'body' of the requests. This means that the url will be http:localhost:3000/calculator/add but the parameters will be sent in the request body. Since you cannot send POST request from your browser without any extension, Postman is a tool that makes it easy to send these requests:
![image](https://user-images.githubusercontent.com/22799292/141351318-5c831d9e-8d56-45dd-a5cd-83d04eb4a430.png)

Now, as far as your nodejs app and how to read parameters from the body, you need to:

1. Enable express to read a request's body, by adding this to the top of your script after you create your app:
```
const app = express();
app.use(express.urlencoded({ extended: true }));
```
2. Then, simply access the request's body instead of the query:
```
const params = req.body;
const values = Object.values(params)
```
Try these steps, experiment with postman if you'd like and let me know how it works out!